datafusion/core/tests/physical_optimizer/partition_statistics.rs

```diff
@@ -488,4 +520,155 @@ mod test {
+            total_byte_size: Precision::Absent,
+            column_statistics: vec![
+                ColumnStatistics {
+                    null_count: Precision::Absent,
+                    max_value: Precision::Exact(ScalarValue::Int32(Some(2))),
+                    min_value: Precision::Exact(ScalarValue::Int32(Some(1))),
+                    sum_value: Precision::Absent,
+                    distinct_count: Precision::Absent,
+                },
+                ColumnStatistics::new_unknown(),
+                ColumnStatistics::new_unknown(),
+            ],
+        };
+
+        let p1_statistics = aggregate_exec_partial.partition_statistics(Some(1))?;
+        assert_eq!(&p1_statistics, &expected_p1_statistics);
+
+        validate_statistics_with_data(
+            aggregate_exec_partial.clone(),
+            vec![
+                ExpectedStatistics::NonEmpty(3, 4, 2),
+                ExpectedStatistics::NonEmpty(1, 2, 2),
+            ],
+            0,
+        )
+        .await?;
+
+        let agg_final = Arc::new(AggregateExec::try_new(
+            AggregateMode::Final,
+            group_by.clone(),
+            aggr_expr.clone(),
+            vec![None],
+            aggregate_exec_partial.clone(),
+            aggregate_exec_partial.schema(),
+        )?);
+
+        let p0_statistics = agg_final.partition_statistics(Some(0))?;
+        assert_eq!(&p0_statistics, &expected_p0_statistics);
+
+        let p1_statistics = agg_final.partition_statistics(Some(1))?;
+        assert_eq!(&p1_statistics, &expected_p1_statistics);
+
+        validate_statistics_with_data(
+            agg_final.clone(),
+            vec![
+                ExpectedStatistics::NonEmpty(3, 4, 2),
+                ExpectedStatistics::NonEmpty(1, 2, 2),
+            ],
+            0,
+        )
+        .await?;
+
+        let empty_table =
+            Arc::new(EmptyExec::new(scan_schema.clone()).with_partitions(2));
+
+        let agg_partial = Arc::new(AggregateExec::try_new(
+            AggregateMode::Partial,
+            group_by.clone(),
+            aggr_expr.clone(),
+            vec![None],
+            empty_table.clone(),
+            scan_schema.clone(),
+        )?);
+
+        let agg_final = Arc::new(AggregateExec::try_new(
+            AggregateMode::Final,
+            group_by.clone(),
+            aggr_expr.clone(),
+            vec![None],
+            agg_partial.clone(),
+            agg_partial.schema(),
+        )?);
+
+        let empty_stat = Statistics {
+            num_rows: Precision::Exact(0),
+            total_byte_size: Precision::Absent,
+            column_statistics: vec![
+                ColumnStatistics::new_unknown(),
+                ColumnStatistics::new_unknown(),
+                ColumnStatistics::new_unknown(),
+            ],
+        };
+        assert_eq!(&empty_stat, &agg_final.partition_statistics(Some(0))?);
+        assert_eq!(&empty_stat, &agg_final.partition_statistics(Some(1))?);
```

Comment:
Is it reasonable to returen two partitions after Agg with `AggregateMode::Final`. In my mind, it should be only one partition.

Reply:
How about also adding `AggregateMode::Final` test

Reply:
> How about also adding `AggregateMode::Final` test

done. PTAL